projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts

```diff
@@ -1,12 +1,101 @@
 import { TestBed } from '@angular/core/testing';
 
 import { CaseSharingStateService } from './case-sharing-state.service';
+import {SharedCase} from "../../models/case-share.model";
+import {UserDetails} from "../../models/user-details.model";
 
 describe('CaseSharingStateService', () => {
+  let service: CaseSharingStateService;
+  let sharedCases: SharedCase[] =[];
   beforeEach(() => TestBed.configureTestingModule({}));
 
+  beforeEach(() => {
+    service = TestBed.get(CaseSharingStateService);
+  });
+
   it('should be created', () => {
-    const service: CaseSharingStateService = TestBed.get(CaseSharingStateService);
     expect(service).toBeTruthy();
+
+  });
+
+  it('should set cases', () => {
+    sharedCases  = [{
+      caseId: '9417373995765133',
+      sharedWith: [],
+      caseTitle: 'Sam Green Vs Williams Lee'
+    }];
+    service.setCases(sharedCases);
+    expect(service).toBeTruthy();
```

Comment:
Validation missing. Validate if case added to service.

Reply:
modified the change by size.